### PR TITLE
I'm selling changes

### PR DESCRIPTION
**Pull request by eswordthecat**

### CODE REVIEW

File: code/game/objects/items/weapons/RCD.dm [hunk 1]
```diff
@@ -145,87 +145,115 @@ RCD
...
 	switch(mode)
 		if(1)
-			if(istype(A, /turf/space))
-				if(useResource(floor_cost, user))
+			if (istype(target, /turf/space))
+				if (useResource(floor_cost, user))
 					user << "Building Floor..."
 					activate()
-					A:ChangeTurf(/turf/simulated/floor/plating/airless)
+					target:ChangeTurf(/turf/simulated/floor/plating/airless)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/RCD.dm]: Casting.

Comment (eswordthecat) [inline code/game/objects/items/weapons/RCD.dm]: Blame old code

File: code/game/objects/items/weapons/RCD.dm [hunk 2]
```diff
@@ -145,87 +145,115 @@ RCD
...
-					if(do_after(user, 40))
-						if(!useResource(decon_cost, user)) return 0
+
+					if (do_after(user, 40))
+						if(!useResource(decon_cost, user))
+							return 0
+
 						activate()
-						A:ChangeTurf(/turf/simulated/floor/plating)
+						target:ChangeTurf(/turf/simulated/floor/plating)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/RCD.dm]: : makes God cry.

File: code/modules/hydroponics/hydro_tray.dm [hunk 3]
```diff
@@ -297,15 +297,15 @@
...
-		if(A.lighting_use_dynamic)
-			light_available = max(0,min(10,T.lighting_lumcount)-5)
-		else
-			light_available =  5
-		if(abs(light_available - seed.ideal_light) > seed.light_tolerance)
-			health -= healthmod
+	var/light_available
+
+	if (areaMaster.lighting_use_dynamic)
+		light_available = max(0, min(10, T.lighting_lumcount) - 5)
```
Comment (ComicIronic) [inline code/modules/hydroponics/hydro_tray.dm]: Clamp this.

File: code/modules/hydroponics/hydro_tray.dm [hunk 4]
```diff
@@ -788,15 +788,14 @@
...
 			var/light_available
-			if(A)
-				if(A.lighting_use_dynamic)
-					light_available = max(0,min(10,T.lighting_lumcount))
-				else
-					light_available =  5
 
-			usr << "The tray's sensor suite is reporting a light level of [light_available] lumens and a temperature of [environment.temperature]K."
+			if (areaMaster.lighting_use_dynamic)
+				light_available = max(0, min(10, T.lighting_lumcount) - 5)
```
Comment (ComicIronic) [inline code/modules/hydroponics/hydro_tray.dm]: Clamp this.

File: code/modules/hydroponics/hydro_tray.dm (hunk 3 above)
Comment (d3athrow) [inline code/modules/hydroponics/hydro_tray.dm]: clamp?

File: code/game/objects/items/weapons/RCD.dm (hunk 1 above)
Comment (ComicIronic) [inline code/game/objects/items/weapons/RCD.dm]: Too late, blaming you.

File: code/modules/hydroponics/hydro_tray.dm (hunk 4 above)
Comment (d3athrow) [inline code/modules/hydroponics/hydro_tray.dm]: clamp?

Comment (eswordthecat) [inline code/modules/hydroponics/hydro_tray.dm]: How i'm thinking about it, but that -5

File: code/modules/hydroponics/hydro_tray.dm (hunk 3 above)
Comment (ComicIronic) [inline code/modules/hydroponics/hydro_tray.dm]: Clamp is just a max(min( (or a min(max() between two values.

File: code/modules/hydroponics/hydro_tray.dm (hunk 4 above)
Comment (ComicIronic) [inline code/modules/hydroponics/hydro_tray.dm]: Clamp(T.lighting_lumcount, 10, 5) - 5 should work, I think.

Comment (eswordthecat) [inline code/modules/hydroponics/hydro_tray.dm]: math is hard

**Pull request closed without merging**